src/global.ts

```diff
@@ -50,36 +54,134 @@ export function setupPythonSupport() {
   setupStatus = "completed";
 }
 
-export async function loadPyodide(artifactsUrl?: string) {
+function getAsyncMemory() {
+  if (
+    "SharedArrayBuffer" in globalThis &&
+    "Atomics" in globalThis &&
+    (globalThis as any)["crossOriginIsolated"] !== false
+  ) {
+    const asyncMemory: AsyncMemory = new AsyncMemory(
+      new SharedArrayBuffer(8 * Int32Array.BYTES_PER_ELEMENT),
+      new SharedArrayBuffer(100)
+    );
+    return asyncMemory;
+  } else {
+    return null;
+  }
+}
+
+export async function loadPyodide(artifactsUrl?: string, workerUrl?: string) {
   if (pyodideLoadSingleton) return pyodideLoadSingleton;
 
   loadingStatus = "loading";
-  const artifactsURL =
-    artifactsUrl ||
-    getPluginOpts().artifactsUrl ||
-    (window as any).pyodideArtifactsUrl ||
-    "https://cdn.jsdelivr.net/pyodide/v0.17.0/full/";
-  pyodideLoadSingleton = (window as any).loadPyodide({ indexURL: artifactsURL }) as Promise<void>;
-  await pyodideLoadSingleton;
-  loadingStatus = "ready";
+  const worker = workerUrl ? new Worker(workerUrl) : new Worker(new URL("pyodide-worker.js", import.meta.url));
+  const asyncMemory = getAsyncMemory();
+  let dataToTransfer: Uint8Array | undefined = undefined;
 
-  // TODO: perhaps we can do this in a cleaner way by passing an output element to runPython or something.
-  (window.pyodide as any).matplotlibHelpers = {
-    createElement: (tagName: string) => {
-      const elem = document.createElement(tagName);
-      if (!CURRENT_HTML_OUTPUT_ELEMENT) {
-        console.log("HTML output from pyodide but nowhere to put it, will append to body instead.");
-        document.querySelector("body")!.appendChild(elem);
-      } else {
-        CURRENT_HTML_OUTPUT_ELEMENT.appendChild(elem);
+  pyodideLoadSingleton = new Promise((resolve, reject) => {
+    // Only the resolve case is handled for now
+    function handleInitMessage(ev: MessageEvent<any>) {
+      if (ev.data && (ev.data as WorkerResponse).type === "initialized") {
+        worker.removeEventListener("message", handleInitMessage);
+        resolve(worker);
+      }
+    }
+    worker.addEventListener("message", handleInitMessage);
+  });
+
+  worker.addEventListener("message", (e) => {
+    if (!e.data || e.data.type === undefined) return;
```

Comment:
If we never expect this to happen, maybe let's console error or warn here? Something like:

console.warn("Pyodide webworker received unexpected message:", e)